tests/4.5/target_teams_distribute_parallel_for/test_target_teams_distribute_parallel_for_private.F90

```diff
@@ -0,0 +1,77 @@
+!===--- test_target_teams_distribute_parallel_for_private.F90 ---------------===//
+!
+! OpenMP API Version 4.5 Nov 2015
+! 
+! This test check for a private variable within a pragma omp target teams 
+! distribute parallel for. We use a private variable within a for loop and 
+! assign it every iteration hoping that we won't get into data races. We do this 
+! multiple times to improve testing. We assign a large number of threads and
+! teams to try to increase parallelism and contention on the privatized variable
+!
+!===-------------------------------------------------------------------------===//
+
+#include "ompvv.F90"
+
+#define N 1024
+
+PROGRAM test_target_teams_distribute_parallel_for_private 
+   USE iso_fortran_env
+   USE ompvv_lib
+   USE omp_lib
+   implicit none
+
+   OMPVV_TEST_OFFLOADING
+
+   OMPVV_TEST_VERBOSE(target_teams_distribute_parallel_for_private() .ne. 0)
+
+   OMPVV_REPORT_AND_RETURN()
+
+CONTAINS
+   INTEGER FUNCTION target_teams_distribute_parallel_for_private()
+      INTEGER, DIMENSION(N) :: a, b, c, d, num_teams, num_threads
+      INTEGER :: privatized, errors, i, j, warning_threads, warning_teams
+      
+      errors = 0
+      j = 0
+
+      DO i = 1, N
+         a(i) = 1
+         b(i) = i
+         c(i) = 2*i
+         d(i) = 0
+         num_teams(i) = -1
+         num_threads(i) = -1
+      END DO
+
+      !$omp target data map(to: a, b, c) map(from: d)
+         !$omp target teams distribute parallel do private(privatized, i)&
+         !$omp& num_threads(OMPVV_NUM_THREADS_DEVICE) num_teams(OMPVV_NUM_TEAMS_DEVICE)
+         DO j = 1, N
+            num_teams(j) = omp_get_num_teams()
+            num_threads(j) = omp_get_num_threads()
+
+            privatized = 0
+            
+            DO i = 1, (a(j) + b(j))
+               privatized = privatized + 1
+            END DO
+            
+            d(j) = c(j) * privatized
+         END DO
+      !$omp end target data
+
+      warning_threads = 0
+      warning_teams = 0
+
+      DO i = 1, N
+         OMPVV_TEST_AND_SET(errors, d(i) .ne. ((1 + i) * 2 * i))
+         warning_teams = warning_teams + num_teams(i)
```

Comment:
Why would you use warning_teams = warning_teams + num_teams(i) if you previously just set warning_teams = 0? (Still correct, just curious)

Reply:
If this was not inside a for loop I would just do 'warning_teams = num_teams(i)', but since it is nested inside I want to update the value of warning teams N number of times